gapis/gfxapi/gles/resources.go

```diff
@@ -203,17 +204,28 @@ func (shader *Shader) SetResourceData(ctx context.Context, at *path.Command,
 	for resource.Accesses[index] > at.Index && index >= 0 {
 		index--
 	}
+
+	to_mutate := len(list.Atoms)
+	if to_mutate > 5 {
+		to_mutate = 5
+	}
+	// Hack until the architecture is part of the header
```

Comment:
I'll have the header CL up for review later today. Do you want to wait for this to be in so you don't have to add more hacks?